### PR TITLE
Install limited amount of modules for UKI Ubuntu

### PR DESCRIPTION
**Pull request by mauromorales**
Not sure if this is the right approach but since it's quite simple to test this with the multi stage dockerfile then I wanted to give it a go

Not an original idea, recommended by Kevin

relates to https://github.com/kairos-io/kairos/issues/2565

### CODE REVIEW

**Review by Itxaka — COMMENTED** [review]
That was fast! But this should be a separated image no? Or are we shipping this by default?

**Comment by mauromorales** [comment]
@Itxaka that's where I'm not sure, I don't see why not ship it like this and if we see that it doesn't work in many places then we go back to the "big" image, but in general I think the space issue will be a concern for any user

**Comment by Itxaka** [comment]
> @Itxaka that's where I'm not sure, I don't see why not ship it like this and if we see that it doesn't work in many places then we go back to the "big" image, but in general I think the space issue will be a concern for any user

I would say, we keep them separated until we can provide the modules/fw into the system with an extra step during install or image build or something no?

Otherwise we are exchanging something that works out of the box for 95% of systems with something that might or might not work out of the box for an unknown percentage of systems with no workaround other than....rebuild the image yourself directly and install those packages.


IMHO provided images by us should be pretty generic and support as much software as possible, even if we have to notice that some EFI implementations will not be able to boot those image, then point to this nano images as an alternative, less supported, more difficult ones.

But that's just my opinion. For sure if someone is gonna consume this, we need to get them in our workflow to test them in case we add a dep like a module that is missing. The rest is up for debate.

**Comment by Itxaka** [comment]
We probably are also winning in the EFI firmware part with smaller images. After all fedora does not ship so much stuff and seems ok...

One for the other lol

**Comment by mauromorales** [comment]
@Itxaka yeah good question, what is more generic, cuz if the image does not fit, then it doesn't matter that it has all the generics. But I don't have a strong opinion, we can go either or. I think this is something that maybe we can offer multiple options, with the least amount of firmware to all the firmware and let the user decide since they are the ones with the hardware

**Comment by Itxaka** [comment]
> @Itxaka yeah good question, what is more generic, cuz if the image does not fit, then it doesn't matter that it has all the generics. But I don't have a strong opinion, we can go either or. I think this is something that maybe we can offer multiple options, with the least amount of firmware to all the firmware and let the user decide since they are the ones with the hardware

Yep indeed, I think it would be good to offer both options.

Full firmware, most of the stuff supported out of the box. Downsides, it may not boot in all hardware.

Lite, should boot in most of the systems out there. Downsides, it may not support all your hardware, might require post install steps to provide extra modules/firmware 


So we cover both ways and have alternatives for the uki stuff.

**Comment by jimmykarily** [comment]
I'm still getting a huge ISO from this branch:

```
~/workspace/kairos/kairos (reduce-size-of-uki-image)*$ du -h build/kairos-ubuntu-24.04-core-amd64-generic-v3.0.4-71-ge24bc2ac.iso
1,2G	build/kairos-ubuntu-24.04-core-amd64-generic-v3.0.4-71-ge24bc2ac.iso
```

am I supposed to pass a flag to earthly or something?

**Comment by mauromorales** [comment]
@jimmykarily yes you need to pass `--BOOTLOADER=systemd-boot`

**Comment by jimmykarily** [comment]
```
$ ~/workspace/kairos/kairos (reduce-size-of-uki-image)*$ earthly +iso --VARIANT=core --FLAVOR=ubuntu --FLAVOR_RELEASE=24.04  --BASE_IMAGE=ubuntu:24.04  --MODEL=generic --FAMILY=ubuntu  --BOOTLOADER=systemd-boot

...

                +iso *failed* | INFO[2024-05-23T06:18:45Z] Starting rsync...                            
                +iso *failed* | DEBU[2024-05-23T06:18:45Z] Running cmd: 'rsync --progress --partial --human-readable --archive --xattrs --acls --exclude=/mnt --exclude=/proc --exclude=/sys --exclude=/dev --exclude=/tmp --exclude=/host --exclude=/run /grub2/ /tmp/enki-iso071332082/iso/' 
                +iso *failed* | INFO[2024-05-23T06:18:45Z] Finished syncing                             
                +iso *failed* | INFO[2024-05-23T06:18:45Z] Finished copying /grub2 into /tmp/enki-iso071332082/iso 
                +iso *failed* | ERRO[2024-05-23T06:18:45Z] No initrd file found                         
                +iso *failed* | ERRO[2024-05-23T06:18:45Z] Could not find kernel and/or initrd          
                +iso *failed* | ERRO[2024-05-23T06:18:45Z] Failed preparing ISO's root tree: No file found with prefixes: [initrd initramfs] 
                +iso *failed* | ERRO[2024-05-23T06:18:45Z] 1 error occurred:
                +iso *failed* | 	* No file found with prefixes: [initrd initramfs]
                +iso *failed* |  
                +iso *failed* | ERROR Earthfile line 544:4
                +iso *failed* |       The command
                +iso *failed* |           RUN /entrypoint.sh --name $ISO_NAME --debug build-iso --squash-no-compression --date=false dir:/build/image --output /build/
                +iso *failed* |       did not complete successfully. Exit code 1

Help: To debug your build, you can use the --interactive (-i) flag to drop into a shell of the failing RUN step: "earthly -i +iso --VARIANT=core --FLAVOR=ubuntu --FLAVOR_RELEASE=24.04 --BASE_IMAGE=ubuntu:24.04 --MODEL=generic --FAMILY=ubuntu --BOOTLOADER=systemd-boot"

...

**Comment by jimmykarily** [comment]
I'm stupid :facepalm: . This is not meant to build an iso (`+iso`) but only a base-image to be passed to enki.

**Review by jimmykarily — APPROVED** [review]

File: images/Dockerfile.ubuntu
Comment (mauromorales) [inline images/Dockerfile.ubuntu:144]:
```suggestion
```

**Pull request merged**
